python/lsst/daf/butler/core/datastoreRecordData.py

```diff
@@ -82,15 +120,29 @@ class DatastoreRecordData:
     datastore.
     """
 
-    refs: List["DatasetIdRef"] = dataclasses.field(default_factory=list)
-    """List of DatasetRefs known to this datastore.
+    dataset_ids: Set[DatasetId] = dataclasses.field(default_factory=set)
+    """List of DatasetIds known to this datastore.
```

Comment:
I'm surprised this field is useful, given that its contents seem like they should be the same as `records.keys()`.  Is that not the case, or perhaps just not _guaranteed_ to be the case?

Reply:
My thinking was that there may not be a guarantee for some future implementation. Thinking about it again, it is already possible to model that case by using an empty dict in `records` for that key. I'll drop `dataset_ids`.